components/ruby/lib/chef-licensing/license_key_fetcher/file.rb

```diff
@@ -14,6 +15,11 @@ class File
       LICENSE_KEY_FILE = "licenses.yaml".freeze
       LICENSE_FILE_FORMAT_VERSION = "4.0.0".freeze
 
+      SUPPORTED_LICENSE_FILE_FORMAT_VERSIONS = [
```

Comment:
we don't need this if you ask me since we already have `LICENSE_FILE_FORMAT_VERSION` to define the current format version and we can simply check if the `Object.const_get(Version::LicenseFile)` exists that should give you the supported versions already

Reply:
yup, agreed! Although I had to use `Object.const_defined?` to achieve the existence check of the class